src/main/scala/dsptools/Chisel3TestersCompatibility.scala

```diff
@@ -0,0 +1,14 @@
+package chisel3.iotesters
```

Comment:
So the goal of a good API is to both prevent unsafe things (like being to add pins at arbitrary points of elaboration) and to provide an intuitive and clean interface that makes sense and isn't brittle. Of course, the latter points are mostly pretty subjective...

- flatten: several reasons this is bad:
  - flatten doesn't make sense on Data, since Data can be either an Aggregate or Element. This is probably more a style issue, since the current implementation isn't incorrect. Though this is constraining in that everything must reduce to Bits, even where it might not make sense (Clock, or even other types we may not have thought of yet).
  - This also breaks individual levels of abstraction, for example you're essentially punching through several layers of Bundles and stuff to get the underlying elements. A better abstraction is to consider things on level at a time. This does require some structural changes, such as having hierarchical directionality (which is planned for the Bindings refactor).
  - Note: I want to deprecate this with the asTypeOf / Bindings refactor. Do you have a use case where you need it that can't be considered at each level (possibly recursively)?
- getDataNames, validName: these are tester internal details. There would be a good argument here on how to make testers extensible, and restructuring may make it possible for you to do what you need without this hack. Reason internal details are hidden is that they aren't part of the API, which gives developers flexibility to change the implementation without breaking user code.
- bigIntToStr: I don't even know why we have this. [BigInt.toString(radix: Int)](http://www.scala-lang.org/api/2.12.0/scala/math/BigInt.html#toString(radix:Int):String) exists.